api/pkg/store/mongo/store.go

```diff
@@ -94,29 +133,9 @@ func (s *Store) ListDevices(ctx context.Context, perPage, page int, filters []mo
 	}
 
 	query := []bson.M{
-		{
-
-			"$lookup": bson.M{
-				"from":         "connected_devices",
-				"localField":   "uid",
-				"foreignField": "uid",
-				"as":           "online",
-			},
-		},
-		{
-			"$lookup": bson.M{
-				"from":         "users",
-				"localField":   "tenant_id",
-				"foreignField": "tenant_id",
-				"as":           "namespace",
-			},
-		},
-		{
-			"$addFields": bson.M{
-				"online":    bson.M{"$anyElementTrue": []interface{}{"$online"}},
-				"namespace": "$namespace.username",
-			},
-		},
+		lookupDevices,
+		lookupUsers,
+		onlineDevices,
```

Comment:
This rework is very nice; it'd be good if it comes before the addition of the multiple filters so it is even easier to follow.